plugins/network/NetFavorites.cpp

```diff
@@ -2,84 +2,86 @@
 #include "Network.hpp"
 #include "NetCommon.hpp"
 #include "NetLng.hpp"
-
-#define SZ_FAVORITES          L"Favorites"
-#define SZ_FAVORITES_SUBKEY   SZ_FAVORITES L"\\%s"
-#if 0
-#define SZ_USERNAME           L"UserName"
-#define SZ_USERPASS           L"Password"
-#endif
-
+#include "guid.hpp"
+#include <PluginSettings.hpp>
+#include <string>
+#include <vector>
+#include <locale>
+
+const wchar_t SZ_FAVORITES[] = L"Favorites";
+const wchar_t SZ_USERNAME[] = L"UserName";
+const wchar_t SZ_USERPASS[] = L"Password";
 static wchar_t szFavProv[] = L"Far Favorites Provider";
 
-BOOL GetFavorites(LPNETRESOURCE pNR, NetResourceList* pList)
+std::vector<std::wstring> SplitPath(const wchar_t* lpRemoteName)
 {
-	NETRESOURCE tmp = {0};
-	tmp.dwDisplayType = RESOURCEDISPLAYTYPE_DOMAIN;
+	// lpRemoteName = Favorite\\share \\machine\share or \\machine
+	// result  {Favorite , machine, share} or {Favorite , machine}
+	// machine name must be in uppercase
+	std::vector<std::wstring> result;
+	result.reserve(3);
+	std::wstring remote_name{lpRemoteName};
+	result.emplace_back(SZ_FAVORITES);
+	if (remote_name.find(SZ_FAVORITES) == 0)
+	{
+		remote_name = remote_name.substr(std::size(SZ_FAVORITES));
+	}
+
+	size_t i = 0;
+	while (remote_name[i] == L'\\')
+	{
+		i++;
+	}
+
+	std::wstring machine_name{remote_name.substr(i)};
+	auto const pos = machine_name.find(L'\\');
+	if (pos != std::string::npos)
+	{
+		auto share_name = machine_name.substr(pos + 1);
+		machine_name = machine_name.substr(0, pos);
+		std::use_facet<std::ctype<wchar_t>>(std::locale()).toupper(
```

Comment:
This only works with the base ascii character set.
I'd prefer delegating *everything* locale-, case- etc. related to the OS, e.g. via [CharUpperBuffW](https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-charupperbuffw).

This is probably not critical in this particular place - machine names are restricted to [A-Z0-9.-] anyway - but still, eventually the code could be copy-pasted elsewhere.